packages/engine-multi/src/worker/thread/runtime.ts

```diff
@@ -38,7 +39,8 @@ export const publish = (
     type,
     threadId,
     processId,
-    ...payload,
+    // ensure the event message is serializable
+    ...JSON.parse(stringify(payload)),
```

Comment:
So this is my solution for now.

It's pretty brute force.

Every event that gets sent through the thread will be serialized with `fast-safe-stringify`, then parsed back into JSON, and then published (Where presumably it's serialized again)